clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreImpl.java

```diff
@@ -92,8 +92,9 @@ public void close() {
     store.close();
   }
 
-  public List<OffHeapChainMap<Long>> getSegments() {
-    return store.getSegments();
+  @Override
+  public List<Set<Long>> getSegmentKeySets() {
+    return store.getSegments().stream().map(OffHeapChainMap::keySet).collect(Collectors.toList());
```

Comment:
Why not simply return an anonymous abstract list that delegates the two methods you have to implement.